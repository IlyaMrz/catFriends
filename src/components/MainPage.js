// separated redux app (app3) for better testing 

import React, { Component} from 'react';
import SearchBox from './SearchBox';
import Scroll from './Scroll';
import ErrorBoundry from './ErrorBoundry';
import CardList from './CardList'
// import CardList from '../CardList';
// import { robots } from './robots';


class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots()
    }

    filterRobots = () => {
        return this.props.robots.filter(robot => {
            return robot.name.toLowerCase().includes(this.props.searchfield.toLowerCase())
        })
    }

    render() {
        const {onSearchChange,isPending} = this.props;

        return isPending ?
            <h1>loading</h1> :
            <div className='tc'>
                <SearchBox searchChange={onSearchChange} />
                <Scroll>
                    <ErrorBoundry>
                        <CardList robots={this.filterRobots()} />
                    </ErrorBoundry>
                </Scroll>

            </div>


    }
}
export default MainPage;
