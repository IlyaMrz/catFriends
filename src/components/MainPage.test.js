import { shallow, mount, render } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';
import '../setupTest';

let wrapper;
beforeEach(()=>{
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchfield: '',
        isPending: false
    }
    wrapper = shallow(<MainPage { ...mockProps } />)
})

it("expect to render App-MainPage component",()=> {
    expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', ()=>{
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 4,
            name: 'Nob',
            email: 'nob@bob.tt'
        }],
        searchfield: '3',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage { ...mockProps2 } />)
    expect(wrapper2.instance().filterRobots()).toEqual([])
})