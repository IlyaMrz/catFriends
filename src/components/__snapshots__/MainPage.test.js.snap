// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render App-MainPage component 1`] = `
<div
  className="tc"
>
  <SearchBox />
  <Scroll>
    <ErrorBoundry>
      <CardList
        robots={Array []}
      />
    </ErrorBoundry>
  </Scroll>
</div>
`;
